codegen/valagasyncmodule.vala

```diff
@@ -261,6 +261,21 @@ public class Vala.GAsyncModule : GtkModule {
 
 		ccode.add_assignment (new CCodeMemberAccess.pointer (data_var, "_async_result"), create_result);
 
+		var full_method_name = new StringBuilder ();
+		full_method_name.append (m.name.has_prefix (".") ? m.name.substring (1) : m.name);
```

Comment:
Interesting -- a method name prefixed with `.`, when is this the case?

Reply:
constructors. Initially i just checked if they where constructors and returned "new", but then i found out constructors can have name
from test finish-name.vala
```public class Foo {
	[CCode (finish_name = "foo_new_end")]
	public async Foo.async () {
	}

	[CCode (finish_name = "foo_bar_end")]
	public async string bar () {
		return "bar";
	}
	[CCode (finish_name = "foo_baz_end", finish_vfunc_name = "baz_end")]
	public virtual async string baz () {
		return "baz";
	}
}```